validator/src/bin/solana-test-validator.rs

```diff
@@ -362,14 +362,16 @@ fn main() {
         .and_then(sol_str_to_lamports);
 
     let (sender, receiver) = unbounded();
-    run_local_faucet_with_port(
-        faucet_keypair,
-        sender,
-        Some(faucet_time_slice_secs),
-        faucet_per_time_cap,
-        faucet_per_request_cap,
-        faucet_addr.port(),
-    );
+    std::thread::spawn(move || {
+        let faucet = Arc::new(std::sync::Mutex::new(Faucet::new(
+            faucet_keypair,
+            Some(faucet_time_slice_secs),
+            faucet_per_time_cap,
+            faucet_per_request_cap,
+        )));
+        let runtime = tokio::runtime::Runtime::new().unwrap();
```

Comment:
I'd spawn a single-thread runtime for this one, rather than the default (which will use num_cpus worth of threads). Or use multi-thread with 2 threads.